src/components/scenes/Fio/FioConnectWalletConfirmScene.tsx

```diff
@@ -27,222 +28,222 @@ export interface FioConnectWalletConfirmParams {
   walletsToDisconnect: FioConnectionWalletItem[]
 }
 
-interface State {
-  acknowledge: boolean
-  connectWalletsLoading: boolean
-  showSlider: boolean
+interface FioConnectWalletConfirmProps extends EdgeAppSceneProps<'fioConnectToWalletsConfirm'> {
+  wallet: EdgeCurrencyWallet
 }
 
-interface StateProps {
-  ccWalletMap: CcWalletMap
-  isConnected: boolean
-}
+export const FioConnectWalletConfirmComponent = (props: FioConnectWalletConfirmProps) => {
+  const { wallet: fioWallet, navigation, route } = props
+  const theme = useTheme()
+  const styles = getStyles(theme)
+  const dispatch = useDispatch()
 
-interface OwnProps extends EdgeAppSceneProps<'fioConnectToWalletsConfirm'> {
-  wallet: EdgeCurrencyWallet
-}
+  const { fioAddressName, walletsToConnect, walletsToDisconnect } = route.params
 
-interface DispatchProps {
-  updateConnectedWallets: (fioAddress: string, ccWalletMap: CcWalletMap) => void
-}
+  const ccWalletMap = useSelector(state => state.ui.fio.connectedWalletsByFioAddress[fioAddressName] ?? {})
+  const isConnected = useSelector(state => state.network.isConnected)
 
-type Props = StateProps & DispatchProps & OwnProps & ThemeProps
+  const [acknowledge, setAcknowledge] = React.useState(false)
+  const [connectWalletsLoading, setConnectWalletsLoading] = React.useState(false)
+  const [showSlider, setShowSlider] = React.useState(true)
 
-export class FioConnectWalletConfirm extends React.Component<Props, State> {
-  state = {
-    acknowledge: false,
-    connectWalletsLoading: false,
-    showSlider: true
-  }
+  const updateConnectedWallets = useHandler((currentFioAddress: string, currentCcWalletMap: CcWalletMap) => {
```

Comment:
This doesn't need to use `useHandler` hook because it's not being used as a prop. It can just be a function declaration or arrow function. This will avoid the need for a unnecessary ref (which the useHandler hook creates).

Alternatively, you could just inline the `dispatch({ ... })` call where each `updateConnectedWallets` is invoked.